internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,104 @@
+package eventstojobs
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+	"github.com/G-Research/armada/pkg/api"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+type EventsToJobService struct {
+	queue                string
+	jobsetid             string
+	jobid                string
+	jobServiceConfig     *configuration.JobServiceConfiguration
+	jobServiceRepository repository.JobServiceRepository
+}
+
+func NewEventsToJobService(
+	queue string,
+	jobsetid string,
+	jobid string,
+	jobServiceConfig *configuration.JobServiceConfiguration,
+	jobServiceRepository repository.JobServiceRepository) *EventsToJobService {
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
+		jobServiceConfig:     jobServiceConfig,
+		jobServiceRepository: jobServiceRepository,
+	}
+}
+
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
+
+	return eventToJobService.StreamCommon(&eventToJobService.jobServiceConfig.ApiConnection, context)
+}
+func (eventToJobService *EventsToJobService) StreamCommon(clientConnect *client.ApiConnectionDetails, ctx context.Context) error {
+	var fromMessageId string
+	conn, connErr := client.CreateApiConnection(clientConnect)
+	eventToJobService.jobServiceRepository.SubscribeJobSet(eventToJobService.jobsetid)
+	if connErr != nil {
+		log.Warnf("Connection Issues with EventClient %v", connErr)
+	}
+	defer conn.Close()
```

Comment:
This is potentially bugged in the broken case; you will not always have a conn to Close in the case where connErr is set.

Reply:
you also close the connection on line 62- I'm not sure if it's legal to close an already closed connection, but it seems a bit wrong to close it twice either way.